### ISSUE TITLE
`set-alias-page.py` generates misleading pages

### ISSUE THREAD

**Issue by acuteenvy** [tldr-pages/tldr#12747]
The [`get_alias_page` function](https://github.com/tldr-pages/tldr/blob/435a0c0a181ced196740a89020425f6fa847afa1/scripts/set-alias-page.py#L128) doesn't do its job properly. This script then works on non-alias pages, and subsequently makes a huge mess by generating misleading pages. It also has some issues with page titles that contain spaces.

Examples: #11365.
Previous runs of this script: 
- #11331, reverted in #11374,
- #9672 ~**not reverted**~, [some pages are affected](https://github.com/tldr-pages/tldr/pull/12734#discussion_r1591648171), partially reverted (for languages without a CODEOWNER, see #12757 for details)

***`set-alias-page.py` should not be used until this is fixed.***

**Comment by blueskyson** [comment]
I just wrote a script with the help of ChatGPT to filter out potentially affected pages:
```python
import os
import re

def filter_files(directory):
    # List to store names of files that meet the criteria
    matching_files = []

    # Iterate through all files in the given directory
    for filename in os.listdir(directory):
        file_path = os.path.join(directory, filename)
        
        # Check if it's a file (and not a directory)
        if os.path.isfile(file_path):
            with open(file_path, 'r', encoding='utf-8') as file:
                lines = file.readlines()
                
                contains_tldr = False
                contains_alias = False
                
                for line in lines:
                    if re.search(r"^`tldr ", line):
                        contains_tldr = True
                    if "This command is an alias of" in line:
                        contains_alias = True
                        
                if contains_tldr and not contains_alias:
                    matching_files.append(filename)
    
    # Print the names of the matching files
    if len(matching_files) > 0:
        print(directory)
        for file in matching_files:
            print(file)
        print()

# Replace 'your_directory_path' with the actual directory path you want to process
repo_path = '/home/lin/Desktop/github/tldr'
filter_files(repo_path + '/pages/osx')
filter_files(repo_path + '/pages/sunos')
filter_files(repo_path + '/pages/openbsd')
filter_files(repo_path + '/pages/android')
filter_files(repo_path + '/pages/freebsd')
filter_files(repo_path + '/pages/windows')
filter_files(repo_path + '/pages/linux')
filter_files(repo_path + '/pages/netbsd')
filter_files(repo_path + '/pages/common')
```
and the results are:
```
/home/lin/Desktop/github/tldr/pages/osx
launchd.md

/home/lin/Desktop/github/tldr/pages/openbsd
pkg.md

/home/lin/Desktop/github/tldr/pages/windows
del.md
wget.md
rmdir.md
gal.md
ri.md
ni.md
rm.md
iwr.md
pwd.md
cd.md
curl.md
clear.md
cls.md
mv.md
mi.md
reg.md
sl.md
gl.md
move.md

/home/lin/Desktop/github/tldr/pages/linux
pkgctl.md
cgroups.md
xbps.md
tailf.md
nmcli.md
eselect.md
distrobox.md
systemd-resolve.md
apx.md
unmount.md
lid.md

/home/lin/Desktop/github/tldr/pages/common
linode-cli.md
ruff.md
transmission.md
ppmnorm.md
ppmtowinicon.md
pgmnorm.md
ppmtojpeg.md
open.md
bundler.md
cups.md
pngtopnm.md
pamfixtrunc.md
lckdo.md
pbmtoicon.md
pcdindex.md
ppmtogif.md
clamav.md
pnmcomp.md
pgmslice.md
todoman.md
pgmedge.md
ykman.md
pnminterp.md
bmptoppm.md
moreutils.md
pnmflip.md
ppmquant.md
pgmcrater.md
pnmtotiff.md
frp.md
just.md
ps-nvm.md
musl-gcc.md
tldr.md
pamrgbatopng.md
pnmarith.md
ripgrep.md
ppmtomap.md
pgmtopbm.md
pgmoil.md
pnmsplit.md
powershell.md
pnmenlarge.md
winicontoppm.md
pbmtox10bm.md
ppmquantall.md
cron.md
pnmcut.md
ppmtotga.md
pnmfile.md
ppmtouil.md
icontopbm.md
ppmbrighten.md
pnmscale.md
gemtopbm.md
pnmtofits.md
```

**Comment by blueskyson** [comment]
I was too careless writing the `set-alias-page` script then 😰

**Comment by Epik-Whale463** [comment]
I suggest updating the **get_alias_page** function with a better regex to help accurately identify valid alias pages and manage titles with spaces.

**Comment by sbrl** [comment]
Yeah, I think updating the script as @Epik-Whale463 suggests is a good plan.

**Comment by sbrl** [comment]
Ref your last comment there, I thought the spec required that e.g. `ip route show` be translated to `ip-route-show` internally?

**Comment by blueskyson** [comment]
@sebastiaanspeck I think the script was not well structured (of course I was not good enough at that time), so now there are too many special cases that are difficult to resolve. I think we should rewrite the script to solve the problems.

**Comment by Managor** [comment]
Fixed in #19859

**Comment by sebastiaanspeck** [comment]
I want to reopen it, when running it for example on Dutch, (`scripts/set-alias-page.py -Sl nl`), it reports it will update over 200 pages, but nothing changes.

**Comment by Managor** [comment]
I think I found the reason for that while playing around to find more stuff to fix.
I added a print statement here:

<img width="703" height="63" alt="Image" src="https://github.com/user-attachments/assets/6d362a15-231a-4458-b464-cf3b9d19aaed" />

I ran `scripts/set-alias-page.py -S -l fi`

And I get this result:

<img width="397" height="130" alt="Image" src="https://github.com/user-attachments/assets/5795ad13-ee25-4717-a83a-ed04e9396f44" />

It seems that the script sets the en template and en reference page to the target language when it finds it in the target language.

I have currently no motivation to fix it as it doesn't block me from doing things, just leaving the info here

**Comment by sebastiaanspeck** [comment]
> I think I found the reason for that while playing around to find more stuff to fix.
> I added a print statement here:
> 
> <img width="703" height="63" alt="Image" src="https://github.com/user-attachments/assets/6d362a15-231a-4458-b464-cf3b9d19aaed" />
> 
> I ran `scripts/set-alias-page.py -S -l fi`
> 
> And I get this result:
> 
> <img width="397" height="130" alt="Image" src="https://github.com/user-attachments/assets/5795ad13-ee25-4717-a83a-ed04e9396f44" />
> 
> It seems that the script sets the en template and en reference page to the target language when it finds it in the target language.
> 
> I have currently no motivation to fix it as it doesn't block me from doing things, just leaving the info here

I would rather see that you fix it, since right now the script generates misleading output, which makes `tldr-maintenance` show Dutch for example has 234 missing alias pages (while they are there).
